compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/JvmInlineClassLowering.kt

```diff
@@ -583,7 +586,7 @@ private class JvmInlineClassLowering(context: JvmBackendContext) : JvmValueClass
                     .single { it is IrSimpleFunction && it.name.asString() == name } as IrFunction
                 val branches = noinlineSubclasses
                     .filter { info ->
-                        info.bottom.declarations.any { it is IrSimpleFunction && it.name.asString() == name && !it.isFakeOverride }
+                        info.bottom.isFunctionDeclared() || info.sealedParents.any { it.owner.isFunctionDeclared() }
```

Comment:
Maybe you could use `overriddenSymbols` to figure out if there is an implementation of a function from `Any` with the correct name?